### ISSUE TITLE
DynaSim automatic linting

### ISSUE THREAD

**Issue by erik-roberts** [DynaSim/DynaSim#296]
List of things to check:
- linkers all linked
- no parameters specified that don't match at least 1 parameter in a mech file
- vary matches a parameter

these are all things that would result in wrong simulation but won't throw error at present (i don't think)

**Comment by erik-roberts** [comment]
expanding on #71

**Comment by kupiqu** [comment]
I agree with @erik-roberts these cannot be considered optimal coding practices.

As for running multiple realizations of stochastic differential equations (issue #71), this is so important that should be a first citizen of DynaSim (using `vary` with a fix label variable specifically for this purpose, e.g. `realization`, `iteration` or whatever) and should be documented how exactly users should do it... would you agree?

I don't think dummy variables are the best option here.

**Comment by erik-roberts** [comment]
when you say first citizen, do you mean there should be a standard variable name always there that can be used and varied? could this be related to varying over a random seed with a random draw for the seed?

**Comment by kupiqu** [comment]
Yes, that's right.

I was also going to say that it would be good if the seed could be set to shuffle or just set it to the iteration number so simulations are easily reproducible, but the seed is saved anyways also when using shuffle, right @jsherfey? If so, we could just use the shuffle

On the other hand, in order to reproduce a set of simulations we need a way to extract all random seeds from them all first, which is tricky...
